### PR TITLE
Bug #4792

### PR DESCRIPTION
**Pull request by stunney**
Adding validation of WixPdb to paired MSI to ensure that the two
actually belong together.

### CODE REVIEW

**Comment by robmen** [comment]
Two very big things:

1. Only check the PackageCode. That will be an exact match.
2. Don't send messages by throwing exceptions. Write a message like all the other code.

**Comment by stunney** [comment]
1.  Agreed
2.  Crap, my bad.

**Comment by stunney** [comment]
Would you recommend that melt still "abort" upon discovering the mismatch or simply serve up a "WARNING" on the console?

**Comment by robmen** [comment]
For WiX v3.x I think it *has* to be a warning since maybe someone is doing this and it is working for them (magically). It *should* become an error in v4.x.

**Comment by stunney** [comment]
Already added the warning string to the resource file.

**Comment by stunney** [comment]
New PR will be up tonight.

**Comment by robmen** [comment]
You don't need a new PR, you can just update your branch and push back to it. That will update this pull request.

**Comment by stunney** [comment]
updates committed

File: src/tools/melt/melt.cs [hunk 1]
```diff
@@ -308,6 +308,30 @@ private static void MeltBinaryTable(Pdb inputPdb, InstallPackage package, string
             }
         }
 
+        private static bool ValidateMSIMatchesPdb(InstallPackage package, Pdb inputPdb)
+        {
+            const string KEY_PRODUCT_CODE = "ProductCode";
```
Comment (heaths) [inline src/tools/melt/melt.cs]: By declaring `const` you're actually causing the string to be allocated multiple times. Either declare as `static readonly` at the class scope or declare here without `const` so the same string instance is used.

Comment (stunney) [inline src/tools/melt/melt.cs]: Can't use static readonly at the class level and maintain the usage of the switch statement.  That said, it is a single check so I can easily switch that to an if statement.

**Comment by stunney** [comment]
Updates based on @heaths feedback committed.

**Pull request merged**